config/accounts.go

```diff
@@ -131,6 +131,18 @@ func (a *AccountGDPR) PurposeEnforced(purpose consentconstants.Purpose) (value,
 	return *a.PurposeConfigs[purpose].EnforcePurpose, true
 }
 
+// PurposeEnforcementAlgo checks the purpose enforcement algo for a given purpose by first
+// looking at the account settings, and if not set there, defaulting to the host configuration.
+func (a *AccountGDPR) PurposeEnforcementAlgo(purpose consentconstants.Purpose) (value string, exists bool) {
+	if a.PurposeConfigs[purpose] == nil {
+		return TCF2FullEnforcement, false
+	}
+	if a.PurposeConfigs[purpose].EnforceAlgo == TCF2FullEnforcement {
+		return TCF2FullEnforcement, true
+	}
```

Comment:
I recommend using Go's response for map lookups and reuse the value response. Something like:
```
c, exists := a.PurposeConfigs[purpose]

if exists {
	return TCF2FullEnforcement, c.EnforceAlgo == TCF2FullEnforcement
}

return TCF2FullEnforcement, false
```

Reply:
Is the first return value anticipating a future PR? It seems that it must always be `TCF2FullEnforcement` in this code.

Reply:
Yeah I confused myself. Originally this PR was just going to be the refactoring part and basic enforcement was going to be introduced in another PR but I ended up including basic enforcement here but never added the basic config option. I'll add it now.